asyncio/task.py

```diff
@@ -21,6 +21,18 @@
 from . import core
 
 
+class CancelledError(BaseException):
+    """Injected into a task when calling `Task.cancel()`"""
+
+    pass
+
+
+class InvalidStateError(Exception):
+    """Can be raised in situations like setting a result value for a task object that already has a result value set."""
+
+    pass
+
+
```

Comment:
You could define these conditionally by testing whether they are in `_asyncio`, using `hasattr()`.  I don't know whether you would then want to define them in `task.py` or `core.py`.

Reply:
I can do that, sure.

To clarify, I was kind of doing this already by the `from _asyncio import CancelledError, InvalidStateError`.  In the case of the `_asyncio` existing but missing the exceptions (because of a mismatched version of CircuitPython?) an `AttributeError` is raised, leading to pulling them from task instead.

Even with that in mind, these should use `hasattr()`?

>  I don't know whether you would then want to define them in task.py or core.py.

Where they live is somewhat inconsequential.  An alternative was that they'd live under an `exceptions.py`.

Reply:
I was thinking these exceptions are defined unconditionally in `task.py` right now, and that they could be conditionalized. But now I'm realizing that `task.py` is not used at all if `_asyncio` exists. Is that right? If so, then I'm off the mark.

Reply:
If `_asyncio` exists, that's right.

With this new change, if `_asyncio` exists but doesn't have the two new errors we'll reach into task.py and grab these two errors.  That does mean that we'll interpret the rest of `task.py` -- which I'm realizing now might be undesirable?

Reply:
> which I'm realizing now might be undesirable?
Yes, it would be a waste of time and RAM space if it gets imported. So maybe just define them inside one of the arms of the `import` try-except, or factor them into `exceptions.py` as you suggested.

Reply:
Updated per request - it's in the import `try-except` because I didn't want to move the `TimeoutError`